core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -89,6 +90,8 @@ protected QueryShardContext initialValue() {
     private final ParseFieldMatcher parseFieldMatcher;
     private final boolean defaultAllowUnmappedFields;
 
+    private TermsLookupFetchService termsLookupFetchService;
```

Comment:
I think I prefer the Client directly here. This service makes you think that there is no client needed, but it just hides that client away :) I think I prefer to have the client here but keep it contained and add the fetch method directly to IndexQueryParsreService, at least for now